vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -114,6 +121,34 @@ private void configureComponent() {
     addCancelListenerForAddVariableDialog();
     addConfirmListenerForAddVariableDialog();
     addListenerForNewVariableEvent();
+    addEditListenerForExperimentalVariables();
+  }
+
+  private void addEditListenerForExperimentalVariables() {
+    experimentalVariablesComponent.subscribeToEditEvent(experimentalVariablesEditEvent -> {
+      var editDialog = ExperimentalVariablesDialog.prefilled(
+          experimentInformationService.getVariablesOfExperiment(experimentId));
+      editDialog.subscribeToCancelEvent(
+          experimentalVariablesDialogCancelEvent -> editDialog.close());
+      editDialog.subscribeToConfirmEvent(experimentalVariablesDialogConfirmEvent -> {
+        deleteExistingExperimentalVariables(experimentId);
+        registerExperimentalVariables(experimentalVariablesDialogConfirmEvent.getSource());
+        editDialog.close();
+        reloadExperimentalVariables();
+      });
+      editDialog.open();
+    });
+  }
+
+  private void reloadExperimentalVariables() {
+    loadExperiment(experimentId);
+  }
+
+  private void deleteExistingExperimentalVariables(ExperimentId experimentId) {
+    var result = deletionService.deleteAllExperimentalVariables(experimentId);
+    result.onError(responseCode -> {
+      throw new ApplicationException(ErrorCode.GENERAL, ErrorParameters.empty());
+    });
```

Comment:
```suggestion
      throw new ApplicationException("variable deletion failed: " + responseCode, ErrorCode.GENERAL, ErrorParameters.empty());
```
otherwise debugging is hard.

For me this exception is thrown so some check previous to this does not catch that there are samples registered for the experiment.
`projects/e8dcd526-6e77-4ce1-b516-b0e2ebc8171d/experiments/39e9bb86-d1ba-4782-a387-3e02505be3de`